plugins/wallpaper.py

```diff
@@ -1,127 +1,73 @@
-    e = requests.get(link).content
-    paea = 'donno.{}'.format(link.split('.')[-1])
-    path_i = os.path.join(down_p, paea)
-    with open(path_i, 'wb') as k:
-        k.write(e)
-    return path_i
+@userge.on_cmd("wall", about={
+    'header': "Search Wallpaper",
+    'flags': {
+        '-l': "Limit of Wallpapers",
+        '-doc': "Send as Documents (Recommended)"
+    },
+    'description': 'Search and Download Hd Wallpaper from Unsplash and upload to Telegram',
+    'usage': "{tr}wall [Query]",
+    'examples': "{tr}wall luffy"})
+async def wall_(msg: Message):
 
+    if os.path.exists("wallpapers/"):
+        shutil.rmtree("wallpapers/", ignore_errors=True)
 
-async def walld(strin: str):
-    if len(strin.split()) > 1:
-        strin = '+'.join(strin.split())
-    url = 'https://wall.alphacoders.com/search.php?search='
-    none_got = ['https://wall.alphacoders.com/finding_wallpapers.php']
-    none_got.append('https://wall.alphacoders.com/search-no-results.php')
-    page_link = 'https://wall.alphacoders.com/search.php?search={}&page={}'
-    resp = requests.get(f'{url}{strin}')
-    if resp.url in none_got:
-        return False
-    if 'by_category.php' in resp.url:
-        page_link = str(resp.url).replace('&amp;', '') + '&page={}'
-        check_link = True
-    else:
-        check_link = False
-    resp = soup(resp.content, 'lxml')
-    try:
-        page_num = resp.find('div', {'class': 'visible-xs'})
-        page_num = page_num.find('input', {'class': 'form-control'})
-        page_num = int(page_num['placeholder'].split(' ')[-1])
-    except Exception:
-        page_num = 1
-    n = randint(1, page_num)
-    if page_num != 1:
-        if check_link:
-            resp = requests.get(page_link.format(n))
-        else:
-            resp = requests.get(page_link.format(strin, n))
-        resp = soup(resp.content, 'lxml')
-    a_s = resp.find_all('a')
-    list_a_s = []
-    tit_links = []
-    r = ['thumb', '350', 'img', 'big.php?i', 'src', 'title']
-    for a_tag in a_s:
-        if all(d in str(a_tag) for d in r):
-            list_a_s.append(a_tag)
-    try:
-        for df in list_a_s:
-            imgi = df.find('img')
-            li = str(imgi['src']).replace('thumb-350-', '')
-            titl = str(df['title']).replace('|', '')
-            titl = titl.replace('  ', '')
-            titl = titl.replace('Image', '')
-            titl = titl.replace('HD', '')
-            titl = titl.replace('Wallpaper', '')
-            titl = titl.replace('Background', '')
-            p = (li, titl)
-            tit_links.append(p)
-    except Exception:
-        pass
-    del list_a_s
-    if len(tit_links) != 0:
-        tit_link = choice(tit_links)
-    else:
-        return False
-    return tit_link
+    limit = min(int(msg.flags.get('-l', 8)), 10)
 
+    if msg.filtered_input_str:
+        qu = msg.filtered_input_str
+        await msg.edit(f"`Seraching Wallpapers for {qu}`")
+        results = requests.get(
```

Comment:
better if you can use `aiohttp`

Reply:
i was offline at that time, you can wait a little